code/game/gamemodes/nuclear/nuclear.dm

```diff
@@ -80,16 +80,34 @@
 				if(synd_mind.current.client)
 					for(var/datum/mind/synd_mind_1 in syndicates)
 						if(synd_mind_1.current)
-							var/I = image('icons/mob/mob.dmi', loc = synd_mind_1.current, icon_state = "synd")
+							var/imageloc = synd_mind_1.current
+							if(istype(synd_mind_1.current.loc,/obj/mecha))
+								imageloc = synd_mind_1.current.loc
+							var/I = image('icons/mob/mob.dmi', loc = imageloc, icon_state = "synd", layer = 13)
 							synd_mind.current.client.images += I
 
 /datum/game_mode/proc/update_synd_icons_added(datum/mind/synd_mind)
 	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/datum/game_mode/proc/update_synd_icons_added() called tick#: [world.time]")
+	if(!synd_mind)
+		return 0
 	spawn(0)
-		if(synd_mind.current)
-			if(synd_mind.current.client)
-				var/I = image('icons/mob/mob.dmi', loc = synd_mind.current, icon_state = "synd")
-				synd_mind.current.client.images += I
+		for(var/datum/mind/synd in syndicates)

```

Comment:
So the icon gets properly added to everyone. That loop exists for all the other antag roles, except for Nuke Ops since whoever wrote it didn't expect the possibility that admins would make new operatives mid-round. It is needed so the icons properly update for everyone in the team.